tests/mocha/toolbox_test.js

```diff
@@ -137,165 +180,384 @@ suite('Toolbox', function() {
+      var newItem = getNonCollapsibleItem(this.toolbox);
+      testHideFlyout(this.toolbox, newItem, newItem);
+    });
+    test('No new item -> Should close flyout', function() {
+      testHideFlyout(this.toolbox, null, null);
+    });
+    test('Select collapsible item -> Should close flyout', function() {
+      var newItem = getCollapsibleItem(this.toolbox);
+      testHideFlyout(this.toolbox,null, newItem);
+    });
+    test('Select selectable item -> Should open flyout', function() {
+      var showFlyoutstub = sinon.stub(this.toolbox.flyout_, 'show');
+      var scrollToStartFlyout = sinon.stub(this.toolbox.flyout_, 'scrollToStart');
+      var newItem = getNonCollapsibleItem(this.toolbox);
+      this.toolbox.updateFlyout_(null, newItem);
+      sinon.assert.called(showFlyoutstub);
+      sinon.assert.called(scrollToStartFlyout);
+    });
+    test('Select non selectable item -> Should do nothing', function() {
+      var showFlyoutstub = sinon.stub(this.toolbox.flyout_, 'show');
+      var hideFlyoutStub = sinon.stub(this.toolbox.flyout_, 'hide');
+      var nonSelectableItem = getSeparator(this.toolbox);
+      this.toolbox.updateFlyout_(null, nonSelectableItem);
+      sinon.assert.notCalled(showFlyoutstub);
+      sinon.assert.notCalled(hideFlyoutStub);
     });
   });
 
   suite('position', function() {
     setup(function() {
-      this.toolbox.init();
+      this.toolbox = getBasicToolbox();
     });
+    teardown(function() {
+      // Don't need to dispose, init never called on basic toolbox.
+      delete this.toolbox;
+    });
+
     function checkHorizontalToolbox(toolbox) {
       chai.assert.equal(toolbox.HtmlDiv.style.left, '0px', 'Check left position');
       chai.assert.equal(toolbox.HtmlDiv.style.height, 'auto', 'Check height');
-      var svgSize = Blockly.svgSize(toolbox.workspace_.getParentSvg());
-      chai.assert.equal(toolbox.HtmlDiv.style.width, svgSize.width + 'px', 'Check width');
-      chai.assert.equal(toolbox.height, toolbox.HtmlDiv.offsetHeight, 'Check height');
+      chai.assert.equal(toolbox.HtmlDiv.style.width, '100%', 'Check width');
+      chai.assert.equal(toolbox.height_, toolbox.HtmlDiv.offsetHeight, 'Check height');
     }
     function checkVerticalToolbox(toolbox) {
-      var svgSize = Blockly.svgSize(toolbox.workspace_.getParentSvg());
-      chai.assert.equal(toolbox.HtmlDiv.style.height, svgSize.height + 'px', 'Check height');
-      chai.assert.equal(toolbox.width, toolbox.HtmlDiv.offsetWidth, 'Check width');
+      chai.assert.equal(toolbox.HtmlDiv.style.height, '100%', 'Check height');
+      chai.assert.equal(toolbox.width_, toolbox.HtmlDiv.offsetWidth, 'Check width');
     }
-    test('Return if tree is not yet initialized', function() {
-      this.toolbox.HtmlDiv = null;
-      this.toolbox.horizontalLayout_ = true;
-      this.toolbox.position();
-      chai.assert.equal(this.toolbox.height, '');
+    test('HtmlDiv is not created -> Should do nothing', function() {
```

Comment:
maybe instead of "should do nothing" you could say "should not resize",  to be more descriptive

Reply:
updated.